packages/vite/src/node/build.ts

```diff
@@ -613,6 +613,43 @@ async function doBuild(
       normalizedOutputs.push(buildOutputOptions(outputs))
     }
 
+    if (libOptions) {
+      const { libFormats } = resolveLibFormats(libOptions)
```

Comment:
I guess this example will be applied. (I think it shouldn't be.)
```js
{
  build: {
    lib: {
      entry: { main: path.resolve(__dirname, 'src/main.js'), sub: path.resolve(__dirname, 'src/sub.js') },
      name: 'MyLib'
    },
    rollupOptions: {
      output: [
        // format is set in rollupOptions.output instead of lib.formats
        { format: 'es' },
        { name: 'MyLib', format: 'umd' }
      ]
    }
  }
}
```

I think we need to check `outputs` instead of `libFormats`. Something like:
```js
const hasUmdOrIifeFormat = outputs.some(output => output.format === 'umd' || output.format === 'iife')
```